### PR TITLE
Routes and API response general rework

### PR DESCRIPTION
**Pull request by p1003**

### CODE REVIEW

**Review by grzanka — DISMISSED** [review]
LGTM

**Comment by ostatni5** [comment]
Do not send 'No content' as a response. Also, field content is supposed to be a dictionary - keep the same format of fields. 
obraz

obraz

**Comment by ostatni5** [comment]
in /sh/status 
Send in response status of the task. Now is ambiguous if the request failed or the task failed. Because 500 and weird response suggest that the request failed.
obraz

**Review by ostatni5 — CHANGES_REQUESTED** [review]
Adress comments

**Comment by ostatni5** [comment]
What is the purpose of the third cookie it is not removed after logout?
obraz

**Comment by p1003** [comment]
> What is the purpose of the third cookie it is not removed after logout? obraz

It is a bug, `token` shouldn't be present, fixed in next commit

**Comment by p1003** [comment]
> in /sh/status Send in response status of the task. Now is ambiguous if the request failed or the task failed. Because 500 and weird response suggest that the request failed. obraz

This is very hard to manage because it is converter Error but you can assume that this isn't error with your request, because I send you back other status codes if informations in it was incorrect/invalid. You get 500 when server itself or simulation is failing.

File: yaptide/celery/tasks.py [hunk 1]
```diff
@@ -115,57 +115,28 @@ def simulation_task_status(task_id: str) -> dict:
 
     result = {
         'status': 'OK',
-        'message': {
-            'state': task.state
+        'content': {
+            'state': task.state,
```
Comment (grzanka) [inline yaptide/celery/tasks.py:121]:
```suggestion
            'state': task.state
```

do we need this comma ?

Comment (p1003) [reply yaptide/celery/tasks.py:121]: @grzanka we do not, but this is convention used everywhere in the project as in converter repository.

Comment (p1003) [reply yaptide/celery/tasks.py:121]: I am not commiting this suggestion

Comment (grzanka) [reply yaptide/celery/tasks.py:121]: ok, no problem

**Comment by grzanka** [comment]
@p1003 I leave the review for @ostatni5 , for me it is good enough

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (grzanka) [inline .vscode/settings.json]: Do we really need to commit that file ?

File: yaptide/routes/utils/decorators.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    def decorator(f):
+        """Determines if the access or refresh token is valid"""
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            token: str = request.cookies.get(
+                'refresh_token' if is_refresh else 'access_token')
+            if not token:
+                raise Unauthorized(description="No token provided")
+            resp: Union[int, str] = decode_auth_token(
+                token=token, is_refresh=is_refresh)
```
Comment (grzanka) [inline yaptide/routes/utils/decorators.py]:
```suggestion
            resp: Union[int, str] = decode_auth_token(token=token, is_refresh=is_refresh)
```

File: .vscode/settings.json (hunk 2 above)
Comment (p1003) [reply .vscode/settings.json]: it should have been in gitignore

**Review by ostatni5 — DISMISSED** [review]
I do not see any issues for now

File: yaptide/routes/utils/decorators.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def requires_auth(is_refresh: bool):
+    """Decorator for auth requirements"""
+    def decorator(f):
+        """Determines if the access or refresh token is valid"""
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            token: str = request.cookies.get(
+                'refresh_token' if is_refresh else 'access_token')
```
Comment (grzanka) [inline yaptide/routes/utils/decorators.py]:
```suggestion
            token: str = request.cookies.get('refresh_token' if is_refresh else 'access_token')
```

**Review by grzanka — APPROVED** [review]
LGTM

**Pull request merged**